docs/reference/vpc-association-policy.md

```diff
@@ -0,0 +1,85 @@
+
+When attaching a policy to a resource, the following restrictions apply:
+
+* A policy can be only attached to `Gateway` resources.
+* The attached resource should exist in the same namespace as the policy resource.
+
+The security Group will not take effect if:
+
+* The targetRef `Gateway` does not exist
+* AssociateWithVpc field set to false
+
+
+**WARNING**
+
+Current VPC Lattice updateServiceNetworkVpcAssociation api have a limitation that it cannot remove all security groups.
+That means, if you have a VpcAssociationPolicy attached to a gateway that already applied security groups, following operations will NOT take effect to remove the security groups:
+* Update the VPCAssociationPolicy to empty the security group ids (even though the updated VPCAssociationPolicy can be accepted by the API server)
+* Delete the VPCAssociationPolicy (even though the VPCAssociationPolicy can be deleted from k8s successfully)
+
+To remove security groups, instead, you should delete VPC Association and then create a new VPC Association without security group ids by following steps:
+1. Update the VPCAssociationPolicy with AssociateWithVpc is false and empty security group ids
+2. Update the VPCAssociationPolicy with AssociateWithVpc is true and empty security group ids
+
+Be cautious to set AssociateWithVpc to false. That can break traffic from the current cluster workloads to the gateway.
+
+
+| Field	                                                                                                                      | Description	                                        |
+|-----------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------|
+| `apiVersion` *string*	                                                                                                      | ``application-networking.k8s.aws/v1alpha1`` 	       |
+| `kind` *string*	                                                                                                            | ``VpcAssociationPolicy``                            |
+| `metadata` [*ObjectMeta*](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#objectmeta-v1-meta)        	 | Kubernetes metadata for the resource.               |
+| `spec` *VpcAssociationPolicySpec*	                                                                                          | Defines the desired state of VpcAssociationPolicy.	 |
+
+
+## VpcAssociationPolicySpec
+
+Appears on: VpcAssociationPolicy
+
+VpcAssociationPolicySpec defines the desired state of VpcAssociationPolicy.
+
+
+
+| Field	                                                                                                     | Description                                                                                                                                                                                                                |
```

Comment:
Perhaps a `Required` column would help with clarifying whether a field is optional